src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,183 @@
+package duke;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import duke.exception.*;
+
+/** Class Duke, the driver class of the program. */
+public class Duke {
+  private Ui ui;
+  private Parser parser;
+  private TaskList taskList;
+  private Storage storage;
+
+  public static void main(String[] args) throws IOException {
+    new Duke().run();
+  }
+
+  /** Duke object creation, initiate Ui, Parser, TaskList, and Storage. */
+  public Duke() {
+    ui = new Ui();
+    parser = new Parser();
+    taskList = new TaskList();
+    storage = new Storage(".//saved-tasks.txt");
+  }
+
+  /**
+   * Driver function of the whole program.
+   *
+   * <p>Initiates greeting. Enter a while loop waiting for commands and call respective handlers.
+   * Exit when a "bye" command is given.
+   *
+   * @throws IOException If Exception is raised during loading and saving.
+   */
+  public void run() throws IOException {
+    storage.loadBaby(taskList, parser);
+    ui.greeting();
+    ui.showList(taskList.getTaskList(), "Tasks loaded from disk:");
+    ui.initPrompt();
+
+    boolean exit = false;
+
+    while (!exit) {
+      try {
+        String longCommand = ui.getCommand();
+        String[] keywords = parser.parseCommand(longCommand);
+        switch (keywords[0]) {
```

Comment:
Great to see the use of the switch statement here to handle multiple cases of the command word!